src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs

```diff
@@ -0,0 +1,612 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.SimplifyPropertyPattern;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.SimplifyPropertyPattern
+{
+    using VerifyCS = CSharpCodeFixVerifier<
+        CSharpSimplifyPropertyPatternDiagnosticAnalyzer,
+        CSharpSimplifyPropertyPatternCodeFixProvider>;
+
+    public class SimplifyPropertyPatternTests
+    {
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsSimplifyPropertyPattern)]
+        public async Task NotInCSharp9()
+        {
+            var code = @"
+using System;
+using System.Reflection;
+
+class C
+{
+    void S(Type t)
+    {
+        if (t is { Assembly: { EntryPoint: { CallingConvention: CallingConventions.Any } } })
+        {
+
+        }
+    }
+}";
+
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsSimplifyPropertyPattern)]
+        public async Task InCSharp10()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+using System;
+using System.Reflection;
+
+class C
+{
+    void S(Type t)
+    {
+        if (t is { [|Assembly:|] { [|EntryPoint:|] { CallingConvention: CallingConventions.Any } } })
```

Comment:
Can the fixer be invoked either on `Assembly:` or `EntryPoint:` in this case (yielding different results)?

Reply:
yup!